### PR TITLE
feat(multiselect): add onMenuChange event

### LINKED ISSUE #7287: Add multiselect onMenuChange event
### Summary

Add an `onMenuChange` event for the multiselect (dropdown) menu, to know if the user has finished their selection. Currently there is only the `onChange` event which can be used to act on any changes.

### Justification

In Watson Knowledge Catalog, we have multiselects, which enables/disables rows based on the selection within the multiselect. So if nothing is selected, the row gets disabled immediately and the dropdown is closed.
See: https://github.ibm.com/wdp-gov/tracker/issues/42209#issuecomment-25014911

wkc-role-change-wrong

You can get it to work, but only with hacky workarounds:
https://codesandbox.io/s/modern-silence-8myhf?file=/index.js

wkc-role-change-correct

### Desired UX and success metrics

No UX changes.

### "Must have" functionality

Add `onMenuChange` event to multiselect (or dropdowns in general) which is triggered if the menu visibility is changed.

### Specific timeline issues / requests

I guess the workaround would work for now, but I should be added to remove any workarounds to stick to the Carbon default

### Available extra resources

What resources do you have to assist this effort?

https://codesandbox.io/s/modern-silence-8myhf?file=/index.js

### PR DESCRIPTION
**Pull request by ConradSchmidt**
Closes #7287

Adds onMenuChange event to multi select

#### Changelog

**New**

- onMenuChange event on multiselect

**Changed**

**Removed**

#### Testing / Reviewing

- Start storybook
- Open multiselect story
- Open the multiselect menu
- See `onMenuChange: (1) [true]`
- Close the multiselect menu
- See `onMenuChange: (1) [false]`

### CODE REVIEW

**Comment by ConradSchmidt** [comment]
Didn't found an event on the `downshift` hook, therefore I created a simple wrapper.
Maybe this should be applied to other dropdowns as well.

**Comment by ConradSchmidt** [comment]
Based on the issues use case, the solution could be something like this (MultiSelect-story.js):
carbon-role-change

```js
export const Default = withReadme(readme, () => {
  const {
    listBoxMenuIconTranslationIds,
    selectionFeedback,
    ...multiSelectProps
  } = props();

  const [hasFocus, setHasFocus] = useState(false);
  const [active, setActive] = useState(false);
  const [selItems, setSelItems] = useState([items[0]]);
  if (!hasFocus && active && selItems.length == 0) setActive(false);

  return (
    <div style={{ width: 300 }}>
      <Checkbox id="active" checked={active} onChange={a => {
        setActive(a);
        if (a) setSelItems([items[0]]);
      }} labelText="Active" />
      <MultiSelect
        {...multiSelectProps}
        items={items}
        itemToString={(item) => (item ? item.text : '')}
        translateWithId={(id) => listBoxMenuIconTranslationIds[id]}
        selectionFeedback={selectionFeedback}
        key={active}
        disabled={!active}
        onMenuChange={setHasFocus}
        onChange={(e) => {setSelItems(e.selectedItems); }}
      />
    </div>
  );
});
```

**Comment by tw15egan** [comment]
@ConradSchmidt Thanks for taking the time to contribute an enhancement! I think that use-case would be a good story to add for this component, would you mind updating the PR to include the code you've provided? 

BTW, I've updated the snapshot tests so it passes CI so you may just need to pull down those changes first, or run `yarn test -u` from the monorepo root

File: packages/react/src/components/MultiSelect/MultiSelect-story.js [hunk 1]
```diff
@@ -88,6 +89,7 @@ const props = () => ({
     'Selecting more items may increase processing time'
   ),
   onChange: action('onChange'),
+  onMenuChange: action('onMenuChange'),
```
Comment (tw15egan) [inline packages/react/src/components/MultiSelect/MultiSelect-story.js:92]: Should we also add this action to the `with change on close story`?

**Review by tw15egan — APPROVED** [review]
A small comment, but otherwise looks good 👍

**Review by dakahn — APPROVED** [review]

**Comment by ConradSchmidt** [comment]
@tw15egan and @dakahn Thanks for reviewing/feedback and merging that quick!

**Comment by tw15egan** [comment]
@ConradSchmidt Thanks for contributing! 🎉

**Pull request merged**